low-level/files/opt/seagate/sspl/setup/resource_map/lr2/server_manifest.py

```diff
@@ -0,0 +1,173 @@
+        self.field_mapping = {
+            'id': 'uid',
+            'class': 'type',
+            'description': 'description',
+            'product': 'product',
+            'serial': 'serial_number',
+            'vendor': 'manufacturer',
+            'part_number': 'part_number',
+            'model_number': 'model_number',
+            'physid': 'physid',
+            'version': 'version'
+        }
+        self.class_mapping = {
+            'disk': 'hw>disks[%s]>%s',
+            'network': 'hw>networks[%s]>%s',
+            'volume': 'hw>volumes[%s]>%s',
+            'bus': 'hw>buses[%s]>%s',
+            'memory': 'hw>memorys[%s]>%s',
+            'processor': 'hw>processors[%s]>%s',
+            'bridge': 'hw>bridges[%s]>%s',
+            'generic': 'hw>generics[%s]>%s',
+            'storage': 'hw>storages[%s]>%s',
+            'input': 'hw>inputs[%s]>%s',
+            'display': 'hw>displays[%s]>%s',
+            'system': 'hw>systems[%s]>%s',
+            'power': 'hw>powers[%s]>%s'
+        }
+        self.kv_dict = {}
+
+    def get_manifest_info(self, rpath):
+        """
+        Fetch manifest information for given rpath.
+        """
+        logger.info(self.log.svc_log(
+            f"Get Manifest data for rpath:{rpath}"))
+        info = []
+        resource_found = False
+
+        # Fetch server manifest information from lshw
+        if "compute" in rpath:
+            info = self.get_server_manifest_info()
+            resource_found = True
+
+        if not resource_found:
+            msg = f"Invalid rpath or manifest provider doesn't have support for'{rpath}'."
+            logger.error(self.log.svc_log(msg))
+            raise ManifestError(errno.EINVAL, msg)
+        return info
+
+    def get_server_manifest_info(self):
+        """Get server manifest information."""
+        server = []
+        cls_res_cnt = {}
+        data, kvs_dst = self.set_lshw_input_data()
+        for kv_key in data.get_keys():
+            if kv_key.endswith('class'):
+                r_spec = data.get(kv_key)
+                if r_spec in cls_res_cnt:
+                    cls_res_cnt[r_spec] += 1
+                else:
+                    cls_res_cnt[r_spec] = 0
+                if r_spec in self.class_mapping.keys():
+                    for field in self.field_mapping.keys():
+                        manifest_key = self.class_mapping[r_spec] % (cls_res_cnt[r_spec],
+                            self.field_mapping[field])
+                        self.map_manifest_server_data(field, manifest_key, data, kv_key)
+        # Adding data to kv
+        kvs_dst.set(self.kv_dict.keys(), self.kv_dict.values())
+        server = self.get_manifest_output_data()
+        return server
+    
+    def set_lshw_input_data(self):
+        """Fetch lshw data and update it into a file for further execution."""
+        kvs_src = None
+        kvs_dst = None
+        proc = subprocess.Popen(['lshw', '-json'], stdout=subprocess.PIPE)
```

Comment:
Please use `SimpleProcess` from `cortx.utils.process` instead of `subprocess`

Reply:
Done